packages/teams-js/src/internal/nestedAppAuth.ts

```diff
@@ -0,0 +1,223 @@
+ * @property {NestedAppAuthMessageEventNames} messageType - The type of the nested app authentication message event.
+ */
+export interface ParsedNestedAppAuthMessageData {
+  messageType: NestedAppAuthMessageEventNames;
+}
+
+/**
+ * @hidden
+ * Interface for a nested app authentication bridge.
+ *
+ * @internal
+ * Limited to Microsoft-internal use
+ *
+ * @interface
+ *
+ * @property {Function} addEventListener - Function to add an event listener to the bridge. Takes an event name and a callback function as parameters.
+ * @property {Function} postMessage - Function to post a message to the bridge. Takes a message string as a parameter.
+ * @property {Function} removeEventListener - Function to remove an event listener from the bridge. Takes an event name and a callback function as parameters.
+ */
+export interface NestedAppAuthBridge {
+  addEventListener: (eventName: string, callback: (response: string) => void) => void;
+  postMessage: (message: string) => void;
+  removeEventListener: (eventName: string, callback: (response: string) => void) => void;
+}
+
+/**
+ * @hidden
+ * Interface for a Window object extended with a nested app authentication bridge.
+ *
+ * @internal
+ * Limited to Microsoft-internal use
+ *
+ * @interface
+ * @extends {Window}
+ *
+ * @property {NestedAppAuthBridge} nestedAppAuthBridge - The nested app authentication bridge associated with the window.
+ */
+export interface NestedAuthExtendedWindow extends Window {
+  nestedAppAuthBridge: NestedAppAuthBridge;
+}
+
+/**
+ * @hidden
+ * Type for handlers in a nested app authentication bridge.
+ *
+ * @internal
+ * Limited to Microsoft-internal use
+ *
+ * @typedef {Object} NestedAppAuthBridgeHandlers
+ *
+ * @property {Function} onMessage - Function to handle a message event. Takes a MessageEvent object and a callback function as parameters. The callback function is called when a message is received.
+ * @property {Function} handlePostMessage - Function to handle posting a message. Takes a message string as a parameter.
+ */
+type NestedAppAuthBridgeHandlers = {
+  onMessage: (evt: MessageEvent, onMessageReceived: (response: string) => void) => void;
+  sendPostMessage: (message: string) => void;
+};
+
+/**
+ * @hidden
+ * Attempt to polyfill the nestedAppAuthBridge object on the given window
+ *
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export function tryPolyfillWithNestedAppAuthBridge(
+  clientSupportedSDKVersion: string,
+  window: Window | null,
+  handlers: NestedAppAuthBridgeHandlers,
+): void {
+  const logger = tryPolyfillWithNestedAppAuthBridgeLogger;
+
+  if (GlobalVars.isFramelessWindow) {
+    logger('Cannot polyfill nestedAppAuthBridge as current window is frameless');
+    return;
+  }
+
+  if (!window) {
+    logger('Cannot polyfill nestedAppAuthBridge as current window does not exist');
+    return;
+  }
+
+  const parsedClientSupportedSDKVersion = (() => {
+    try {
+      return JSON.parse(clientSupportedSDKVersion);
```

Comment:
Could you please confirm what the value of 'clientSupportedSDKVersion' would be? I want to validate 
```
parsedClientSupportedSDKVersion.supports?.nestedAppAuth
```

Reply:
That's what you have defined here: https://office.visualstudio.com/ISS/_git/metaos-hub-sdk/pullrequest/2481040?_a=files&path=/packages/metaos-hub-sdk/src/Services/MessageHandling/app-message-handler.ts

Reply:
The runtime config of _teams-js_ defines that the app developers can call at runtime via the teams-js APIs. teams-js runtime config is validated using `isSupported`. 

What I'm parsing in the above case is the configuration I get from the HubSDK. So if the hub has defined NAA service, then I'd get `{}` else I'd get `undefined`.